### PR TITLE
Add copy button

### LINKED ISSUE #1004: Add Copy Button

### PR DESCRIPTION
**Pull request by byrichardpowell**
### WHY are these changes introduced?

Fixes #1004 

### WHAT is this pull request doing?

Adds a copy button to copy the preview link.  As part of this work we remove some Polaris stuff since we want to move away from Polaris.  Specifically:

1. Add a new Icon component.  This is a simplified version of Polaris's Icon component that matches our requirements.  To support this we add a VisuallyHidden component.
2. Add a new IconButton component.  This is what we use for the copy button
3. Add [React toastify](https://www.npmjs.com/package/react-toastify) and remove everything to do with Polaris's Toast.  I decided to use a 3P dependency here because Toast can be quite complex.  We can always revisit this.

### How to test your changes?

1. `pnpm install && pnpm build`
2. `cd fixtures/app`
3. `pnpm dev`
4. Open the dev console link the CLI outputs
5. Do the copy buttons work correctly?

### Post-release steps

None

### Measuring impact
How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/ui-extensions-dev-console/src/App.tsx [hunk 1]
```diff
@@ -42,6 +45,22 @@ function App() {
           <AppProvider i18n={enTranslations}>
             <Layout>
               <Routes />
+              <ToastContainer
```
Comment (matteodepalo) [inline packages/ui-extensions-dev-console/src/App.tsx:48]: Minor, but have you considered extracting this in a component and adding it below the `<main>{children}</main>` line in the `Layout` component? Tipically `App.tsx` is quite lean and just a collection of providers/components.

File: packages/ui-extensions-dev-console/src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  accessibilityLabel: string
+}
+
+export function Icon({source, accessibilityLabel}: IconProps) {
+  const SourceComponent = source
+
+  return (
+    <span className={styles.Icon}>
+      <VisuallyHidden>{accessibilityLabel}</VisuallyHidden>
+      <SourceComponent className={styles.Svg} focusable="false" aria-hidden="true" />
```
Comment (matteodepalo) [inline packages/ui-extensions-dev-console/src/components/Icon/Icon.tsx:19]: Would this work if `source` is a string?

**Review by matteodepalo — APPROVED** [review]
Looking good, just a couple of minor comments.

File: packages/ui-extensions-dev-console/src/components/Icon/Icon.tsx (hunk 2 above)
Comment (byrichardpowell) [reply packages/ui-extensions-dev-console/src/components/Icon/Icon.tsx:19]: Not right now. I simplified the use case assuming that we wouldn't need it and we can add it back easily if we do.  Thoughts?

**Pull request merged**